source/common/router/rds_impl.cc

```diff
@@ -43,6 +43,7 @@ RdsRouteConfigProviderImpl::RdsRouteConfigProviderImpl(
     const std::string& stat_prefix, ThreadLocal::SlotAllocator& tls,
     RouteConfigProviderManagerImpl& route_config_provider_manager)
     : runtime_(runtime), cm_(cm), tls_(tls.allocateSlot()),
+      cluster_name_(rds.config_source().api_config_source().cluster_name()[0]),
```

Comment:
Yep, TODO for me works, also tag with https://github.com/lyft/envoy/issues/1170.